Classes/Controller/HaikuController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace T3docs\Examples\Controller;
+
+use T3docs\Examples\Domain\Repository\HaikuRepository;
+use TYPO3\CMS\Core\Service\FlexFormService;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+/**
+ * Class FalExampleController
+ */
+class HaikuController
+{
+    /**
+     * The back-reference to the mother cObj object set at call time
+     */
+    public ContentObjectRenderer $cObj;
+    public array $conf = [];
+    public StandaloneView $view;
+
+    public function __construct(
+        protected readonly HaikuRepository $haikuRepository,
+    ) {
+    }
+
+    /**
+     * This setter is called when the plugin is called from UserContentObject (USER)
+     * via ContentObjectRenderer->callUserFunction().
+     */
+    public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
+    {
+        $this->cObj = $cObj;
+    }
+
+    public function main(string $content, array $conf): string
```

Comment:
As third argument the request is given, so later you can use the PSR-7 request object to retrieve the query parameters instead of using $_REQUEST.

See: https://docs.typo3.org/m/typo3/reference-typoscript/11.5/en-us/ContentObjects/UserAndUserInt/Index.html#userfunc

Reply:
Good catch!